invenio/modules/knowledge/admin.py

```diff
@@ -0,0 +1,194 @@
+##
+## Invenio is free software; you can redistribute it and/or
+## modify it under the terms of the GNU General Public License as
+## published by the Free Software Foundation; either version 2 of the
+## License, or (at your option) any later version.
+##
+## Invenio is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Invenio; if not, write to the Free Software Foundation, Inc.,
+## 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Knowledge Flask Blueprint."""
+
+import os
+
+from flask import request
+
+from invenio.base.i18n import _
+from invenio.ext.admin.views import ModelView
+from invenio.ext.sqlalchemy import db
+from invenio.modules.search.models import Collection
+
+from .forms import KnowledgeForm, DynamicKnowledgeForm, WrittenAsKnowledgeForm, \
+    TaxonomyKnowledgeForm, KnwKBRVALForm
+from .models import KnwKB, KnwKBRVAL
+
+
+def KnwKBRVAL_id_knwKB_formatter(view, context, model, name):
+    """Formatter for the id_knwKB: return the Knowledge name instead."""
+    return model.kb.name
+
+
+def Knowledge_kbtype_formatter(view, context, model, name):
+    """Formatter for the kbtype: return the Knowledge name instead."""
+    value = model.kbtype
+    return next((key for key, val in KnwKB.KNWKB_TYPES.items() if val == value),
+                None)
+
+
+class KnwKBRVALAdmin(ModelView):
+
+    """Flask-Admin module to manage Knowledge Base Mappings."""
+
+    # form_columns = ('m_key', 'm_value', 'kb')
+    form = KnwKBRVALForm
+
+    column_list = ['id_knwKB', 'm_key', 'm_value']
+    column_labels = dict(
+        id_knwKB='Knowledge',
+        m_key='Map From',
+        m_value='To')
+    column_formatters = dict(
+        id_knwKB=KnwKBRVAL_id_knwKB_formatter
+    )
+    column_sortable_list = ('m_key', 'm_value', 'id_knwKB')
+    column_searchable_list = ('m_key', 'm_value')
+    column_default_sort = 'id_knwKB'
+    # TODO add filter for id_knwKB!
+    column_filters = ['m_key']
+
+    def __init__(self, app, *args, **kwargs):
+        """Constructor.
+
+        :param app: flask application
+        """
+        super(KnwKBRVALAdmin, self).__init__(*args, **kwargs)
+
+
+class KnowledgeAdmin(ModelView):
+
+    """Flask-Admin module to manage knowledge."""
+
+    _can_create = True
+    _can_edit = True
+    _can_delete = True
+
+    acc_view_action = 'cfgbibknowledge'
+    acc_edit_action = 'cfgbibknowledge'
+    acc_delete_action = 'cfgbibknowledge'
+
+    form = KnowledgeForm
+
+    column_list = ('name', 'kbtype', 'description')
+    column_formatters = dict(
+        kbtype=Knowledge_kbtype_formatter
+    )
+    column_filters = ('kbtype',)
+    column_choices = {
+        'kbtype': [(v, k) for (k, v) in KnwKB.KNWKB_TYPES.iteritems()]
```

Comment:
actually, now do not really need this field, I'm waiting that the issue #2690 is resolved because `column_choices` I think it is useful only in the presence of filters in the list.

http://flask-admin.readthedocs.org/en/latest/api/mod_model/#flask.ext.admin.model.BaseModelView.column_choices